diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index 2cb8978..685fd7f 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -90,8 +90,9 @@
 	func(DHCP, dhcp, na)
 
 #define CONFIG_BOOTCOMMAND \
-	"load mmc 0:1 ${loadaddr} uEnv.txt;"	\
+	"if load mmc 0:1 ${loadaddr} uEnv.txt; then echo \"booting from SD card\"; else load mmc 1:1 ${loadaddr} uEnv.txt; echo \"booting from eMMC\"; fi;"	\
 	"env import -t ${loadaddr} ${filesize};"	\
+	"if printenv initrd_file; then setenv fdtfile \"am335x-boneblack.dtb\"; echo \"legacy image\"; fi;"	\
 	"run uenvcmd;"
 
 #include <config_distro_bootcmd.h>
